test/framework/machines.go

```diff
@@ -22,11 +22,51 @@ import (
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/test/framework/internal/log"
 	"sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/cluster-api/util/patch"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+//UpdateNodeDrainTimeoutInMachineDeploymentInput is the input for UpdateNodeDrainTimeoutInMachineDeployment
+type UpdateNodeDrainTimeoutInMachineDeploymentInput struct {
+	MachineDeployments         []*clusterv1.MachineDeployment
+	NodeDrainTimeout           *metav1.Duration
+	ClusterProxy               ClusterProxy
+	Cluster                    *clusterv1.Cluster
+	WaitForMachinesToBeUpdated []interface{}
+}
+
+//UpdateNodeDrainTimeoutInMachineDeployment updates the nodeDrainTimeout field in the machinedeployment
+//and wiat until all machines are updated
+func UpdateNodeDrainTimeoutInMachineDeployment(ctx context.Context, input UpdateNodeDrainTimeoutInMachineDeploymentInput) {
```

Comment:
Instead of patching the node drain timeout here, how about having a new flavor with node drain timeout?
Looks like at the beginning of the node drain test, timeout is patched right away so we can have a cluster-template-nodedrain directory under `data/infrastructure-docker` that will be added to the cluster-template using `kustomize`.